### PR TITLE
Fix the accessor attribute for Atom feedUrl link

### LINKED ISSUE #8: Does not parse <content:encoded> and <guid> elements (among others)
Trying to use rss-parser with Instant Article feed the following RSS results in incomplete data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<rss xmlns:content="http://purl.org/rss/1.0/modules/content/" xmlns:dc="http://purl.org/dc/elements/1.1/" version="2.0">
  <channel>
    <title>Instant Article Test</title>
    <link>https://localhost:8000</link>
    <description>1, 2, 1, 2… check the mic!</description>
    <language>en</language>
    <pubDate>Fri, 13 May 2016 15:14:05 GMT</pubDate>
    <dc:date>2016-05-13T15:14:05Z</dc:date>
    <dc:language>en</dc:language>
    <item>
      <title>My first Instant Article</title>
      <link>https://localhost:8000</link>
      <description>Lorem ipsum</description>
      <content:encoded>&lt;b&gt;Lorem&lt;/b&gt; ipsum</content:encoded>
      <pubDate>Wed, 04 May 2016 06:53:45 GMT</pubDate>
      <guid>https://localhost:8000</guid>
      <dc:creator>tobi</dc:creator>
      <dc:date>2016-05-04T06:53:45Z</dc:date>
    </item>
  </channel>
</rss>
```

This is the result in JSON; note the missing `guid` and `content:encoded` data:

```json
{
  "feed": {
    "entries": [
      {
        "title": "My first Instant Article",
        "link": "https://localhost:8000",
        "pubDate": "Wed, 04 May 2016 06:53:45 GMT",
        "content": "Lorem ipsum",
        "contentSnippet": "Lorem ipsum"
      }
    ],
    "title": "Instant Article Test",
    "description": "1, 2, 1, 2… check the mic!",
    "link": "https://localhost:8000"
  }
}
```

### PR DESCRIPTION
**Pull request by andjosh**
Ignore most of these commits. This mainly just fixes #8 complaint about `feedUrl`.

### CODE REVIEW

**Comment by andjosh** [comment]
Also now adds support for parsing podcast feeds.

File: index.js [hunk 1]
```diff
@@ -77,16 +101,35 @@ var parseRSS1 = function(xmlObj, callback) {
...
+  })
   var items = channel.item;
   (items || []).forEach(function(item) {
     var entry = {};
     ITEM_FIELDS.forEach(function(f) {
       if (item[f]) entry[f] = item[f][0];
     })
+    if (item.enclosure) {
+        entry.enclosure = item.enclosure[0].$;
+        entry.itunes = {};
```
Comment (bobby-brennan) [inline index.js]: Let's not create an empty itunes obj unless we know there will be itunes fields. I'd put

```js
entry.itunes = entry.itunes || {}
```
inside the forEach below.

File: index.js [hunk 2]
```diff
@@ -77,16 +101,35 @@ var parseRSS1 = function(xmlObj, callback) {
...
+    }
+    PODCAST_TOP_FIELDS.forEach(function(f) {
```
Comment (bobby-brennan) [inline index.js]: Should be `PODCAST_ITEM_FIELDS`?

File: index.js [hunk 3]
```diff
@@ -77,16 +101,35 @@ var parseRSS1 = function(xmlObj, callback) {
 var parseRSS2 = function(xmlObj, callback) {
   var json = {feed: {entries: []}};
   var channel = xmlObj.rss.channel[0];
-  if (channel['atom:link']) json.feed.feedUrl = channel['atom:link'][0].href;
+  if (channel['atom:link']) json.feed.feedUrl = channel['atom:link'][0].$.href;
   TOP_FIELDS.forEach(function(f) {
     if (channel[f]) json.feed[f] = channel[f][0];
   })
+  if (channel['itunes:owner']) {
```
Comment (bobby-brennan) [inline index.js]: If possible, maybe we can do a top-level check to see if it's an itunes feed and extract all the itunes logic into a separate function

**Comment by bobby-brennan** [comment]
LGTM, thanks for the contribution!

Can you just run

`WRITE_GOLDEN=true npm test`

to update the tests?  It would be great to get an itunes test case in there as well.

**Comment by andjosh** [comment]
Will fix!

**Comment by andjosh** [comment]
Tests updated, test added, method extracted!

**Comment by bobby-brennan** [comment]
Thanks for the contribution!

**Comment by andjosh** [comment]
No problem. Just make sure to release a new version as well!

**Comment by lantrix** [comment]
Awesome work. I had done a simple patch to handle just the enclosure component of an RSS feed as I had needed it, but this is so much more in depth.

**Pull request merged**